cmd/engine/engine.go

```diff
@@ -124,6 +124,9 @@ var (
 	providers            []string
 
 	log *logrus.Entry
+
+	// clouditorVersion is used for the clouditorVersion, linked during the build process
+	clouditorVersion string
```

Comment:
would it make sense to check if this is empty and if yes, then use the git commit sha info from the runtime.BuildInfo?

Reply:
I don't know. We can do that. But I think if we do not get the tag info, we probably won't get the commit hash either. Except when we forget to tag.

Reply:
You always get the the commit hash in `vcs.revision` in https://pkg.go.dev/runtime/debug#BuildSetting

Reply:
Ok this is beyond stupid. If you issue `go build cmd/engine/engine.go` from the root, `vcs.revision` will be empty. If you issue `cd cmd/engine && go build` it correctly sets the revision. Somehow this is documented in https://pkg.go.dev/cmd/go@go1.18#hdr-Relative_import_paths

Reply:
Some additional issues to watch: https://github.com/golang/go/issues/50603

Reply:
`go build ./cmd/engine` also works to produce the git hash, which apparently should be the "correct" way of doing it anyway. 

In this case we still need to inject the official version via `ldflags` and then merge the information at some point.

Reply:
Getting the information from git is not that easy. Maybe we should just create a file with all the important information during the release creation and commits via Github Actions.